lineman/app/components/legacy_trial_expired_modal/legacy_trial_expired.coffee

```diff
@@ -0,0 +1,23 @@
+shownToGroup = {}
+angular.module('loomioApp').factory 'LegacyTrialExpiredModal', (ModalService, AppConfig)  ->
+
+  appropriateToShow: (group, user) ->
+    !shownToGroup[group.id] and
```

Comment:
If the logic is never shared anywhere else I think it feels better to keep it all in the same place.